crates/wit-component/src/linking/metadata.rs

```diff
@@ -89,6 +89,7 @@ impl TryFrom<&FuncType> for FunctionType {
 pub struct GlobalType {
```

Comment:
@dicej I forget if I asked this already, but do you remember why this module duplicates some of the type structures of wasmparser/wasm-encoder?  Would it be possible to migrate this module to using wasmparser directly or wasm-encoder directly for example?

(no need to change anything in this PR yourself @abrown)

Reply:
I don't totally remember, but I think I wanted a `ValueType` that only represented the types we actually support in this module (i.e. no V128 and no reference types).  I.e. I wanted unsupported variants to be unrepresentable.  I'm not necessarily opposed to reusing types from another crate, though.